### PR TITLE
APP-1010: new claim should be shown after closing claims flow

### PR DESCRIPTION
**Pull request by StylianosGakis**

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt [hunk 1]
```diff
@@ -52,6 +53,13 @@ class HomeFragment : Fragment(R.layout.home_fragment) {
             marketManager
         )
 
+        binding.swipeToRefresh.apply {
+            setOnRefreshListener {
+                model.load()
```
Comment (hugokallstrom) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt:58]: You want to set isRefreshing = true here right? And when data is loaded set it to false.

**Review by hugokallstrom — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt [hunk 2]
```diff
@@ -39,6 +39,7 @@ class HomeFragment : Fragment(R.layout.home_fragment) {
     override fun onResume() {
         super.onResume()
         loggedInViewModel.onScroll(scroll)
+        model.load()
```
Comment (hugokallstrom) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt:42]: Will this trigger a reload everytime we switch to this tab?

File: app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt [hunk 3]
```diff
@@ -175,7 +175,7 @@ class EmbarkActivity : BaseActivity(R.layout.activity_embark) {
     private fun setupToolbarMenu(
         progressToolbar: MaterialProgressToolbar,
         hasToolTips: Boolean,
-        isLoggedIn: Boolean
+        isLoggedIn: Boolean,
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/embark/ui/EmbarkActivity.kt:178]: 😍

File: app/src/main/java/com/hedvig/app/feature/home/data/HomeRepository.kt [hunk 4]
```diff
@@ -12,7 +12,7 @@ class HomeRepository(
     private val apolloClient: ApolloClient,
     private val localeManager: LocaleManager,
 ) {
-    fun home() = apolloClient
+    fun homeQueryFlow() = apolloClient
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/data/HomeRepository.kt:15]: I'm personally not a fan of what's called the hungarian notation (https://en.wikipedia.org/wiki/Hungarian_notation), i.e. including type information in the function/variable name. What are your thoughts on the subject? 🙂

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt [hunk 5]
```diff
@@ -52,10 +54,17 @@ class HomeFragment : Fragment(R.layout.home_fragment) {
             marketManager
         )
 
+        binding.swipeToRefresh.apply {
+            setOnRefreshListener {
+                model.load()
+            }
+        }
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt:61]:
```suggestion
        binding.swipeToRefresh.setOnRefreshListener {
            model.load()
        }
```

no need for an apply when it's a single statement 😉

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeViewModel.kt [hunk 6]
```diff
@@ -29,50 +31,44 @@ abstract class HomeViewModel : ViewModel() {
 
     // TODO Fetch address change in progress
     protected val _addressChangeInProgress = MutableLiveData("")
+
     val data: LiveData<Triple<ViewState?, PayinStatusQuery.Data?, String?>> = combineTuple(
         _homeData,
         _payinStatusData,
         _addressChangeInProgress
-    )
+    ).distinctUntilChanged()
```
Comment (hedvigoscar) [inline app/src/main/java/com/hedvig/app/feature/home/ui/HomeViewModel.kt:39]: 💥

**Review by hedvigoscar — COMMENTED** [review]
Don't we need to manually invalidate the cache? Or perhaps do a `NETWORK_ONLY` fetch and write to the cache? 🤔

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt (hunk 5 above)
Comment (StylianosGakis) [reply app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt:61]: Indeed!

File: app/src/main/java/com/hedvig/app/feature/home/data/HomeRepository.kt (hunk 4 above)
Comment (StylianosGakis) [reply app/src/main/java/com/hedvig/app/feature/home/data/HomeRepository.kt]: I agree, it's just that `homeQuery` was already taken and just `home()` really messed up my understanding whenever I saw it and didn't know what it was doing so I felt the urge to change it into anything 😅
This was resolved in a future commit anyway now.

File: app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt (hunk 2 above)
Comment (StylianosGakis) [reply app/src/main/java/com/hedvig/app/feature/home/ui/HomeFragment.kt]: Fixed to only trigger on VM instantiation and when getting back the result of the embark activity

File: app/src/main/java/com/hedvig/app/feature/claims/ui/pledge/HonestyPledgeBottomSheet.kt [hunk 7]
```diff
@@ -28,21 +31,29 @@ class HonestyPledgeBottomSheet : BottomSheetDialogFragment() {
...
+            "claims",
+            getString(R.string.CLAIMS_HONESTY_PLEDGE_BOTTOM_SHEET_BUTTON_LABEL)
+        )
+    }
+
     companion object {
         const val TAG = "HonestyPledgeBottomSheet"
 
-        fun newInstance(): HonestyPledgeBottomSheet = HonestyPledgeBottomSheet()
+        fun newInstance(
```
Comment (hugokallstrom) [inline app/src/main/java/com/hedvig/app/feature/claims/ui/pledge/HonestyPledgeBottomSheet.kt:55]: Neat

**Review by hugokallstrom — APPROVED** [review]

**Pull request merged**